File: remote-agents-dotnet/flows/claude-only.cs
#:project ../RemoteAgents/RemoteAgents.csproj
// flows/claude-only.cs
//
// Baseline flow. Hand it a project name and a prompt; it spins up Claude
// inside a PTY against that project's directory, captures whatever
// changed, and writes a session transcript. No validation, no review,
// no git.
//
// Usage:
//   dotnet run flows/claude-only.cs <project> "<prompt>"
//
// Parity reference: remote-agents/orchestrator/flows/claude-only.mjs

using RemoteAgents.Agents;
using RemoteAgents.Events;
using RemoteAgents.Primitives;
using RemoteAgents.Sessions;

const string FLOW_NAME = "claude-only";

if (args.Length < 2)
{
    Console.Error.WriteLine($"Usage: dotnet run flows/{FLOW_NAME}.cs <project> \"<prompt>\"");
    Environment.ExitCode = 2;
    return;
}

var projectName = args[0];
var userPrompt = string.Join(' ', args[1..]).Trim();

await SubscriptionGuard.CheckAsync();

var projectDir = ProjectRegistry.Resolve(projectName);
var session = Session.Start(new StartSessionRequest(
    ProjectDir: projectDir,
    ProjectName: projectName,
    UserPrompt: userPrompt,
    FlowName: FLOW_NAME));

Console.WriteLine($"[{session.Id}]");
Console.WriteLine($"  flow:    {FLOW_NAME}");
Console.WriteLine($"  project: {projectName} ({projectDir})");
Console.WriteLine($"  prompt:  {userPrompt}");
Console.WriteLine();

var sink = new CompositeSink(
    new ConsoleSink(),
    new JsonlSink(session.TranscriptFile),
    new ProviderJsonlIngestSink(session.Dir, projectDir));

try
{
    var before = FsDiff.Snapshot(projectDir);

    var claude = new ClaudeAgent { Name = "claude", Sink = sink };
    var result = await claude.RunAsync(new AgentRunRequest(
        Prompt: userPrompt,
        SessionId: null,
        ProjectDir: projectDir));

    // Forensic dumps — useful while PTY timings are still being tuned.
    await File.WriteAllTextAsync(Path.Combine(session.Dir, "claude-raw.txt"), result.RawOutput);
    await File.WriteAllTextAsync(Path.Combine(session.Dir, "claude-text.txt"), result.Text);

    var after = FsDiff.Snapshot(projectDir);
    var diff = FsDiff.Diff(before, after);

    Console.WriteLine();
    Console.WriteLine($"Claude session: {result.SessionId}");
    Console.WriteLine($"Files changed:  {diff.Changed.Count}");
    Console.WriteLine($"Files added:    {diff.Added.Count}");
    Console.WriteLine($"Files removed:  {diff.Removed.Count}");
    foreach (var f in diff.All) Console.WriteLine($"  - {f}");

    session.End("done");
    Console.WriteLine();
    Console.WriteLine($"Transcript: {session.Dir}");
}
catch (Exception ex)
{
    Console.Error.WriteLine($"[{FLOW_NAME}] FAILED: {ex}");
    session.End("failed", failureReason: ex.Message);
    Environment.ExitCode = 1;
}

File: remote-agents/orchestrator/flows/claude-only.mjs
// flows/claude-only.mjs
//
// The baseline flow. Hand it a project name and a prompt; it spins up Claude
// inside a PTY against that project's directory, captures whatever changed,
// and writes a session transcript. No validation, no review, no git.
//
// Usage:
//   agents run claude-only <project> "<prompt>"
//   node flows/claude-only.mjs <project> "<prompt>"

import fs from 'node:fs';
import path from 'node:path';
import {
  runClaude,
  requireSubscription,
  resolveProject,
  startSession,
  endSession,
  appendEvent,
  snapshotFiles,
  diffFiles,
} from '../src/index.js';

const FLOW_NAME = 'claude-only';

const [, , projectName, ...promptWords] = process.argv;
const userPrompt = promptWords.join(' ').trim();

if (!projectName || !userPrompt) {
  console.error(`Usage: agents run ${FLOW_NAME} <project> "<prompt>"`);
  process.exit(2);
}

requireSubscription();

const projectDir = resolveProject(projectName);
const session = startSession({
  flowName: FLOW_NAME,
  projectName,
  projectDir,
  userPrompt,
});

console.log(`[${session.id}]`);
console.log(`  flow:    ${FLOW_NAME}`);
console.log(`  project: ${projectName} (${projectDir})`);
console.log(`  prompt:  ${userPrompt}`);
console.log();

appendEvent(session, { kind: 'snapshot-before' });
const before = await snapshotFiles(projectDir);

appendEvent(session, { kind: 'claude-start', prompt: userPrompt });
const result = await runClaude({ prompt: userPrompt, projectDir });
appendEvent(session, {
  kind: 'claude-end',
  sessionId: result.sessionId,
  exitCode: result.exitCode,
  textLength: result.text.length,
});

// Forensic dump — useful while we tune PTY timings; remove later if noisy.
fs.writeFileSync(path.join(session.dir, 'claude-raw.txt'), result.rawOutput);
fs.writeFileSync(path.join(session.dir, 'claude-text.txt'), result.text);

const after = await snapshotFiles(projectDir);
const diff = diffFiles(before, after);
appendEvent(session, { kind: 'diff', ...diff });

console.log();
console.log(`Claude session: ${result.sessionId ?? '(not captured)'}`);
console.log(`Files changed:  ${diff.changed.length}`);
console.log(`Files added:    ${diff.added.length}`);
console.log(`Files removed:  ${diff.removed.length}`);
if (diff.all.length > 0) {
  console.log();
  for (const f of diff.all) console.log(`  - ${f}`);
}

endSession(session, {
  result: 'done',
  filesChanged: diff.all.length,
  claudeSessionId: result.sessionId,
});

console.log();
console.log(`Transcript: ${session.dir}`);
